From 4df3de220987cf22db698e88f308894cc3a34ef4 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Mon, 17 Nov 2025 13:47:11 +0100
Subject: [PATCH] pinctrl: stm32: fix warning for uninitialized variable

GCC incorrectly reports that the variable 'domain' could be used
uninitialized.

To prevent the compile warning, initialize it to NULL.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I717fe449bbe1eb256bec19dc8dc145b7378c7916
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/509723
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/pinctrl/stm32/pinctrl-stm32.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/pinctrl/stm32/pinctrl-stm32.c b/drivers/pinctrl/stm32/pinctrl-stm32.c
index 5d2f66a03db4..4734977b5bcf 100644
--- a/drivers/pinctrl/stm32/pinctrl-stm32.c
+++ b/drivers/pinctrl/stm32/pinctrl-stm32.c
@@ -1822,6 +1822,7 @@ static struct irq_domain *stm32_pctrl_get_irq_domain(struct stm32_pinctrl *pctl)
 		 * Each entry has 3 values: <&phandle 0 0>
 		 */
 		i = DIV_ROUND_UP(i, 3 * STM32_GPIO_IRQ_LINE * sizeof(u32));
+		domain = NULL;
 
 		while (i--) {
 			ret = of_irq_parse_one(np, i * STM32_GPIO_IRQ_LINE, &out_irq);
-- 
2.53.0

